src/components/input/input.ts

```diff
@@ -452,7 +460,19 @@ export default class SlInput extends ShoelaceElement implements ShoelaceFormCont
               'input--no-spin-buttons': this.noSpinButtons
             })}
           >
-            <slot name="prefix" part="prefix" class="input__prefix"></slot>
+            <slot
+              name="prefix"
+              part="prefix"
+              class="${classMap({
+                'input__prefix': true,
+                'input__prefix--currency': !!this.currency,
```

Comment:
instead of adding all these specific classes you should be able to follow what `input--pill` does

Reply:
or could follow what `password-toggle` and `clearable` does which looks like a completely separate if statement from the prefix/suffix slots

Reply:
I would recommend doing what Anne suggested and following the pattern that `pill` uses, which should end up cleaner and with less CSS. We probably also want to override the prefix slot completely if the `currency` attribute is true, because otherwise someone could pass in a prefix or suffix _in addition_ to the currency symbols.